eng/pipelines/templates/stages/archetype-spec-gen-sdk.yml

```diff
@@ -203,6 +193,21 @@ extends:
                 $optional_params
             displayName: 'Generate SDK'
 
+          - task: Powershell@2
+            inputs:
+              filePath: $(SdkRepoDirectory)/eng/common/scripts/Save-Package-Properties.ps1
+              arguments: >
+                -ServiceDirectory $(GeneratedSDK.ServiceName)
```

Comment:
We will need to consider cases where a spec PR containing changes to multiple service folders. In this case, the logic to get the whole service name list might need to reside in the `spec-gen-sdk-runner` tool.

Reply:
Can the `Generate SDK` generate for various SDKs in one run? Even then won't the pipeline run be tired to one service since it's using one distinct config path to trigger SDK generation?

Reply:
For manual trigger scenario, it's true that the run is tied to only one service. However, for both spec PR trigger and batch run scenarios, the pipeline run would touch multiple services. For API view creation, I don't have much concern on the batch run scenario. But we need to support the spec PR scenario.

Reply:
I'm not sure ServiceName and ServiceDirectory are actually the same here. We don't have a great way to set that up. What part of the PackageInfo files does APIView need? Perhaps we have that in the output of our generation step and can avoid the PackageInfo creation.

Reply:
[save-package-properties.yml](https://github.com/Azure/azure-sdk-tools/blob/7a14d60819a9baa6e1ba64a92cd2f21858f9f2f5/eng/common/pipelines/templates/steps/save-package-properties.yml#L2) actually uses `ServiceDirectory`
 not `ServiceName` so I think I should just call the variable ServiceDirectory since that's what it actually is https://github.com/Azure/azure-sdk-tools/blob/7a14d60819a9baa6e1ba64a92cd2f21858f9f2f5/tools/spec-gen-sdk/src/automation/workflowPackage.ts#L121-L122

Reply:
> For manual trigger scenario, it's true that the run is tied to only one service. However, for both spec PR trigger and batch run scenarios, the pipeline run would touch multiple services. For API view creation, I don't have much concern on the batch run scenario. But we need to support the spec PR scenario.

Ok, that mean we need to output a list of service directories.